runtime/src/accounts_db/geyser_plugin_utils.rs

```diff
@@ -83,29 +83,33 @@ impl AccountsDb {
         let slot_stores = slot_stores.read().unwrap();
         let mut accounts_to_stream: HashMap<Pubkey, StoredAccountMeta> = HashMap::default();
         let mut measure_filter = Measure::start("accountsdb-plugin-filtering-accounts");
+        let mut previous_write_version = None;
         for (_, storage_entry) in slot_stores.iter() {
-            let mut accounts = storage_entry.all_accounts();
-            let account_len = accounts.len();
-            notify_stats.total_accounts += account_len;
-            accounts.drain(..).into_iter().for_each(|account| {
+            let accounts = storage_entry.accounts.account_iter();
+            let mut account_len = 0;
+            accounts.for_each(|account| {
+                account_len += 1;
+                if let Some(previous_write_version) = previous_write_version {
+                    assert!(previous_write_version < account.meta.write_version);
```

Comment:
If the write_version is already set to 0 always in this implementation, would not this cause a panic?

Reply:
The change to make it non-zero has already been merged. But, this indicates we don't have a test testing this code path.

Reply:
Talked to Jeff -- we do not have write_version as 0 yet. I will add a test case to catch this.

Reply:
catch it if/when we refactor things to no longer store write version. geyser has some constraints that we need to make sure we're testing.